In [53]:
import math


## Se Dp é fornecido:
Dp = [55*10**-6,40*10**-6,25*10**-6,10*10**-6] #m
rhof = 997 #kg/m³ agua
rhop = 2650 #kg/m³
g = 9.81 #m/s²
mi = 0.890*10**-3 #Pa.s
phi = 1# Esfericidade supondo que seja igual a 1
Q =  0.001 #m³/s 


for dp in Dp:
    
    CdRe_2 = (4/3)*(rhof*(rhop-rhof)*g*(dp)**3)/(mi**2)
    
    # Particulas não esfericas
    
    #K1 = 0.843*math.log10(phi/0.065)
    #K2 = 5.31-4.88*phi
    #Re = ((K1*CdRe_2/24)+(CdRe_2/K2))**(-1/1.2)
    
    # Particulas esfericas
    
    Re = ((CdRe_2/24)**-0.95+(CdRe_2/0.43)**(-0.95/2))**(-1/0.95)
    
    if Re<0.5:
        print(f"Diâmetro de Particula: {dp:.6f}")
        print(f"Regime de Stokes: {Re:.6f}")
        K1=1
        vt = K1*(dp**2)*(rhop-rhof)*g/(18*mi)
        print(f"Velocidade Terminal: {vt:.6f} m/s")
    elif 0.5<Re<1000:
        print(f"Diâmetro de Particula: {dp:.6f}")
        print(f"Regime de Transição: {Re:.6f}")
        vt = (mi*Re)/(dp*rhof)
        print(f"Velocidade Terminal: {vt:.6f} m/s")
    else:
        print(f"Diâmetro de Particula: {dp:.6f}")
        print(f"Regime de Newton: {Re:.6f}")
        K2=0.43
        vt = math.sqrt((4*dp*(rhop-rhof)*g)/(3*K2*rhof))
        print(f"Velocidade Terminal: {vt:.6f} m/s")
   
    
    # Determinando o diametro de cada reator:
    
    D = math.sqrt(4*Q/(math.pi*vt))
    print(f"Diâmetro do elutriador: {D:.6f} m \n")


Diâmetro de Particula: 0.000055
Regime de Stokes: 0.176204
Velocidade Terminal: 0.003062 m/s
Diâmetro do elutriador: 0.644841 m 

Diâmetro de Particula: 0.000040
Regime de Stokes: 0.069455
Velocidade Terminal: 0.001620 m/s
Diâmetro do elutriador: 0.886656 m 

Diâmetro de Particula: 0.000025
Regime de Stokes: 0.017320
Velocidade Terminal: 0.000633 m/s
Diâmetro do elutriador: 1.418650 m 

Diâmetro de Particula: 0.000010
Regime de Stokes: 0.001127
Velocidade Terminal: 0.000101 m/s
Diâmetro do elutriador: 3.546626 m 



## Água

**Elutriador 1:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 55 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.003062 m/s |
| Diâmetro do Elutriador | 0.6448 m |

**Elutriador 2:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 40 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.001619 m/s |
| Diâmetro do Elutriador | 0.8866 m |

**Elutriador 3:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 25 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.0006326 m/s |
| Diâmetro do Elutriador | 1.4186 m |

**Elutriador 4:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 10 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.0001012 m/s |
| Diâmetro do Elutriador | 3.5466 m |



## Ar
**Elutriador 1:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 55 µm |
| Regime | Transição |
| Velocidade Terminal | 0.2076 m/s |
| Diâmetro do Elutriador | 0.07831 m |

**Elutriador 2:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 40 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.1255 m/s |
| Diâmetro do Elutriador | 0.1007 m |

**Elutriador 3:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 25 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.04903 m/s |
| Diâmetro do Elutriador | 0.1611 m |

**Elutriador 4:**

| Parâmetro | Valor |
|-----------|-------|
| Dp | 10 µm |
| Regime | Stokes |
| Velocidade Terminal | 0.007845 m/s |
| Diâmetro do Elutriador | 0.4028 m |

